vtStorAtaProtocol/Platform/Linux/ProtocolAtaPassThrough.cpp

```diff
@@ -27,164 +30,135 @@ namespace vtStor
 namespace Protocol
 {
 
-    const vtStor::U8 FEATURE_REGISTER_OFFSET = 0;
-    const vtStor::U8 ERROR_REGISTER_OFFSET = 0;
-    const vtStor::U8 COUNT_REGISTER_OFFSET = 1;
-    const vtStor::U8 LBA_LOW_REGISTER_OFFSET = 2;
-    const vtStor::U8 LBA_MID_REGISTER_OFFSET = 3;
-    const vtStor::U8 LBA_HIGH_REGISTER_OFFSET = 4;
-    const vtStor::U8 DEVICE_REGISTER_OFFSET = 5;
-    const vtStor::U8 COMMAND_REGISTER_OFFSET = 6;
-    const vtStor::U8 STATUS_REGISTER_OFFSET = 6;
-    const vtStor::U8 RESERVED_REGISTER_OFFSET = 7;
-    
-   /* eErrorCode cAtaPassThrough::IssueCommand( const DeviceHandle& Handle, std::shared_ptr<const cBufferInterface> Essense, std::shared_ptr<cBufferInterface> DataBuffer )
-    {
-        eErrorCode errorCode = eErrorCode::None;
+eErrorCode cAtaPassThrough::IssueCommand(const DeviceHandle& Handle, std::shared_ptr<const cBufferInterface> Essense, std::shared_ptr<cBufferInterface> DataBuffer)
+{
+    eErrorCode errorCode = eErrorCode::None;
 
-        cBufferFormatter bufferFormatter = cBufferFormatter::Reader(Essense);       
-        
-        switch (bufferFormatter.GetHeader().Format)
-        {
-            case 1:
-            {
-                cEssenseAta1 essense = cEssenseAta1::Reader(Essense);
-                
-                InitializePassThroughDirect(
-                    essense.GetCommandCharacteristics(),
-                    essense.GetTaskFileExt(),
-                    essense.GetTaskFile(),
-                    DataBuffer,
-                    60 //TODO: allow configurable timeout
-                    );
-                  
-                U32 bytesReturned = 0;
-                errorCode = IssuePassThroughDirectCommand(Handle, bytesReturned);
-            } break;
-
-            default:
-                errorCode = eErrorCode::FormatNotSupported;
-                break; 
-        }        
-
-        return(errorCode);
-    }
-*/
-   /* void cAtaPassThrough::InitializePassThroughDirect( const StorageUtility::Ata::sCommandCharacteristic& CommandCharacteristics, const StorageUtility::Ata::uTaskFileRegister& PreviousTaskFile, const StorageUtility::Ata::uTaskFileRegister& CurrentTaskFile, std::shared_ptr<cBufferInterface> DataBuffer, U32 TimeoutValueInSeconds )
+    cBufferFormatter bufferFormatter = cBufferFormatter::Reader(Essense);
+
+    switch (bufferFormatter.GetHeader().Format)
+    {
+    case 1:
     {
-        m_AtaPassThrough.Length = sizeof( ATA_PASS_THROUGH_DIRECT );
-        m_AtaPassThrough.DataTransferLength = CommandCharacteristics.DataTransferLengthInBytes;
-        m_AtaPassThrough.TimeOutValue = TimeoutValueInSeconds;
-        if (nullptr != DataBuffer)
+        cEssenseAta1 essense = cEssenseAta1::Reader(Essense);
+
+        InitializePassThroughDirect(essense.GetCommandCharacteristics(), essense.GetTaskFileExt(), essense.GetTaskFile(), DataBuffer, 60);
+
+        if (IOCTL_SG_IO_ERROR == ioctl(Handle.Handle, SG_IO, &m_ATAPassThrough))

```

Comment:
You should put these code lines in a function separately, something likes IssueAtaPassThrough() as an example.